src/CMakeLists.txt

```diff
@@ -35,58 +35,87 @@ else()
     message(FATAL_ERROR "Operating system undefined.")
 endif()
 
-find_package(OpenSSL REQUIRED)
-
-include(GenerateExportHeader)
-
-add_library(mgclient-static STATIC ${mgclient_src_files})
-set_target_properties(mgclient-static PROPERTIES
-        OUTPUT_NAME mgclient)
-target_compile_definitions(mgclient-static PUBLIC MGCLIENT_STATIC_DEFINE)
-target_include_directories(mgclient-static
-        PRIVATE
-        "${PROJECT_SOURCE_DIR}/src"
-        PUBLIC
-        "${PROJECT_SOURCE_DIR}/include"
-        "${CMAKE_CURRENT_BINARY_DIR}"
-        "${OPENSSL_INCLUDE_DIR}")
-target_link_libraries(mgclient-static
-        PRIVATE
-        ${OPENSSL_LIBRARIES} project_options project_c_warnings)
-if(MGCLIENT_ON_WINDOWS)
-    target_link_libraries(mgclient-static PUBLIC ws2_32)
+if(NOT EMSCRIPTEN)
+  find_package(OpenSSL REQUIRED)
 endif()
 
-add_library(mgclient-shared SHARED ${mgclient_src_files})
-set_target_properties(mgclient-shared PROPERTIES
-        OUTPUT_NAME mgclient
-        SOVERSION ${mgclient_SOVERSION}
-        C_VISIBILITY_PRESET hidden)
-target_include_directories(mgclient-shared
-        PRIVATE
-        "${PROJECT_SOURCE_DIR}/src"
-        PUBLIC
-        "${PROJECT_SOURCE_DIR}/include"
-        "${CMAKE_CURRENT_BINARY_DIR}"
-        "${OPENSSL_INCLUDE_DIR}")
-target_link_libraries(mgclient-shared PRIVATE ${OPENSSL_LIBRARIES})
-if(MGCLIENT_ON_WINDOWS)
-    target_link_libraries(mgclient-shared PUBLIC ws2_32)
-endif()
+if(EMSCRIPTEN)
+    list(APPEND mgclient_src_files mgwasm.c)
+    add_executable(mgclient ${mgclient_src_files})
+    set_target_properties(mgclient PROPERTIES LINK_FLAGS "-o mgclient.js -s ASYNCIFY=1 -s MODULARIZE -s EXPORT_NAME=\"load_mgclient\" --shared-memory --no-entry -s USE_PTHREADS=1 -s WEBSOCKET_SUBPROTOCOL=\"binary\" -s EXPORTED_FUNCTIONS='_mg_init, _mg_finalize, _mg_session_params_make, _mg_session_params_destroy, _mg_session_params_set_host, _mg_session_params_set_port, _mg_session_params_set_sslmode, _mg_session_error, _mg_session_run, _mg_session_fetch, _mg_session_pull, _mg_connect, _mg_client_version, _mg_result_row' -s EXPORTED_RUNTIME_METHODS=\"ccall, cwrap, getValue, setValue, UTF8ToString\"")
+
+    target_compile_definitions(mgclient PUBLIC MGCLIENT_STATIC_DEFINE)
```

Comment:
`MGCLIENT_STATIC_DEFINE` is only used in `mgclient-export.h`, which is not used for EMSCRIPTEN, therefore I think this is not necessary.